include/gz/math/graph/Edge.hh

```diff
@@ -332,9 +336,19 @@ namespace graph
   };
 
   /// \brief An invalid directed edge.
+  // Deprecated in favor of NullEdge().
   template<typename E>
   DirectedEdge<E> DirectedEdge<E>::NullEdge(
     VertexId_P(kNullId, kNullId), E(), 1.0, kNullId);
+
+  /// \brief An invalid edge.
+  template<typename E, typename EdgeType>
+  EdgeType &NullEdge()
+  {
+    static auto e = std::make_unique<EdgeType>(
+      VertexId_P(kNullId, kNullId), E(), 1.0, kNullId);
+    return *e;
```

Comment:
looking at [Material::Predefined()](https://github.com/gazebosim/gz-math/blob/0ebc06e80adb5d3359957dd0d9b6b5e9d08013c2/src/Material.cc#L98-L119), I wonder if the unique pointer should be made and released from inside a static lambda function?

Reply:
I think it would be better to use `gz::utils::NeverDestroyed` (e.g. https://github.com/gazebosim/sdformat/blob/f05f4e7ad1a6784f9ff1a6c1b362191677baa70d/src/Types.cc#L149). However, the fact that this function returns a non-const reference could be problematic since anyone can modify the value.

Reply:
regarding the non-const reference, I think the behavior in gz-math7 and earlier with static variables has a similar issue right? I tried changing the `NullEdge` and `NullVertex` functions to return a const reference, but lots of `Graph.hh` functions that currently return non-const references like to return references to `NullEdge` or `NullVertex` and fail to compile. This seems like a bigger issue with the API that we could open an issue about.

Reply:
trying again in https://github.com/gazebosim/gz-math/pull/612